### PR TITLE
Issue #3627 Only renew session id when spnego authentication is fully complete

### PR DESCRIPTION
**Pull request by janbartel**
#3627

Spnego authentication can have multiple client/server round trips, during which time the server needs to keep some info about the in-flight authentication process. This info is kept in a http session. Prior to this PR, after the first client/server round trip we would change the session id and mark it as being secure.  Thus, whilst changing the session id is good, it should not happen until the last request of the authentication sequence is processed when the user is actually authenticated.

### CODE REVIEW

File: jetty-security/src/main/java/org/eclipse/jetty/security/authentication/ConfigurableSpnegoAuthenticator.java [hunk 1]
```diff
@@ -98,6 +99,25 @@ public void setAuthenticationDuration(Duration authenticationDuration)
     {
         _authenticationDuration = authenticationDuration;
     }
+    
+    
+
```
Comment (sbordet) [inline jetty-security/src/main/java/org/eclipse/jetty/security/authentication/ConfigurableSpnegoAuthenticator.java]: Remove unnecessary empty lines.

File: jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java [hunk 2]
```diff
@@ -33,6 +33,14 @@
...
 import org.eclipse.jetty.util.log.Logger;
 
+/**
+ * @author janb
+ *
+ */
+/**
+ * @author janb
+ *
+ */
```
Comment (sbordet) [inline jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java]: Remove these author tags.

File: jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java [hunk 3]
```diff
@@ -53,8 +61,21 @@ public void prepareRequest(ServletRequest request)
...
+     * determine when the UserIdentity IS fully authenticated and renew the session id.
+     * 
+     * @param username the username of the client to be authenticated
+     * @param password the user's credential
+     * @param servletRequest the inbound request that needs authentication
+     * @return
+     */
     public UserIdentity login(String username, Object password, ServletRequest servletRequest)
     {
+
```
Comment (sbordet) [inline jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java]: Remove unnecessary newline.

File: jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java [hunk 4]
```diff
@@ -53,8 +61,21 @@ public void prepareRequest(ServletRequest request)
         //empty implementation as the default
     }
 
+    /**
+     * If the UserIdentity is not null after this method calls LoginService.login, it
```
Comment (sbordet) [inline jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java]: Use javadoc's `@{link ... }` to reference `LoginService.login()` method.

Comment (janbartel) [reply jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java]: Done

File: jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java (hunk 3 above)
Comment (janbartel) [reply jetty-security/src/main/java/org/eclipse/jetty/security/authentication/LoginAuthenticator.java]: Strewth.

**Pull request merged**